db/config.hh

```diff
@@ -387,6 +387,7 @@ public:
     named_value<uint64_t> max_memory_for_unlimited_query_hard_limit;
     named_value<uint32_t> reader_concurrency_semaphore_serialize_limit_multiplier;
     named_value<uint32_t> reader_concurrency_semaphore_kill_limit_multiplier;
+    named_value<int> maintenance_reader_concurrency_semaphore_count_limit;
```

Comment:
I wonder what would happen if someone sets `--maintenance_reader_concurrency_semaphore_count_limit=-1` ?

Reply:
They will block repair/streaming from making any progress. Same if set to 0.

Reply:
OK, fair enough. I think that since nobody objected for seven days we're good to go :)